src/poll.warden.cpp

```diff
@@ -57,13 +57,13 @@ void felspar::io::poll_warden::run_until(felspar::coro::coroutine_handle<> coro)
         } else if (pr > 0) {
             continuations.clear();
             for (auto events : iops) {
-                if (events.revents & POLLIN) {
+                if (events.revents & (POLLIN | POLLERR | POLLNVAL | POLLPRI)) {
                     auto &reads = requests[events.fd].reads;
                     continuations.insert(
                             continuations.end(), reads.begin(), reads.end());
                     reads.clear();
                 }
-                if (events.revents & POLLOUT) {
+                if (events.revents & (POLLOUT | POLLERR | POLLNVAL | POLLPRI)) {
```

Comment:
Do you expect any `POLLPRI` events that will unblock a writer?

Reply:
It's a good point. I think I'll remove it from here, but leave it on the read side -- worst case I guess it could lead to a spurious wake up (which will just go back to sleep), and best case it could stop a lock up for somebody who is using one of these APIs and waiting on a read for it.

Reply:
Hmm, re-reading the man page, it doesn't say that the POLLPRI is only ever in `revents` so maybe it's only seen in response to adding it to `events`, and right now I don't support an API for that. Maybe I just remove both of them.